oauth2_http/java/com/google/auth/mtls/X509Provider.java

```diff
@@ -0,0 +1,127 @@
+package com.google.auth.mtls;
+
+import com.google.api.client.util.SecurityUtils;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.security.KeyStore;
+import java.util.Locale;
+
+public class X509Provider {
```

Comment:
We don't have the equivalent for `@InternalApi` here. Can you add some strongly worded comments above regarding this being internalapi and not meant for users (no backwards compatibility guarantee and the like)?

Reply:
bumping this comment

Reply:
Added a comment explaining that it is internal and warning that we will not support its backward compat.